src/zenml/integrations/mlflow/mlflow_utils.py

```diff
@@ -0,0 +1,159 @@
+#  Copyright (c) ZenML GmbH 2021. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+import os
+from typing import Any, Callable, Optional, Type
+
+import mlflow  # type: ignore
+from mlflow import set_experiment
+from mlflow.tracking import set_tracking_uri  # type: ignore
+
+from zenml.io.utils import get_global_config_directory
+from zenml.pipelines.base_pipeline import BasePipeline
+from zenml.steps import BaseStep
+
+
+def local_mlflow_backend() -> str:
+    """Returns the local mlflow backend inside the global zenml directory"""
+    local_mlflow_backend_uri = os.path.join(
+        get_global_config_directory(), "mlruns"
+    )
+    if not os.path.exists(local_mlflow_backend_uri):
+        os.mkdir(local_mlflow_backend_uri)
+    return local_mlflow_backend_uri
+
+
+def setup_mlflow(
+    backend_store_uri: Optional[str] = None, experiment_name: str = "default"
+) -> None:
+    """Setup all mlflow related configurations. This includes specifying which
+    mlflow tracking uri should b e used and which experiment the tracking
+    will be associated with.
+
+    Args:
+        backend_store_uri: The mlflow backend to log to
+        experiment_name: The experiment name under which all runs will be
+                         tracked
+
+    """
+    # TODO [HIGH]: Implement a way to get the mlflow token and set
+    #  it as env variable at MLFLOW_TRACKING_TOKEN
+    if not backend_store_uri:
+        backend_store_uri = local_mlflow_backend()
+
+    os.environ["MLFLOW_TRACKING_URI"] = backend_store_uri
+    set_tracking_uri(backend_store_uri)
```

Comment:
Aren't these two equivalent ways of configuring the tracking URI ? Or does the env var override the `set_tracking_uri` call ?